nan.h

```diff
@@ -1654,12 +1658,15 @@ class Callback {
       Execute(progress);
   }
 
-  void SendProgress_(const char *data, size_t size) {
-    char *new_data = new char[size];
-    memcpy(new_data, data, size);
+  void SendProgress_(const T *data, size_t size) {
+    T *new_data = new T[size];
+    {
+      T *it = new_data;
+      std::copy(data, data + size, it);
+    }
```

Comment:
@bnoordhuis I assume you intended this? 

With that everything should be addressed cc @kkoopa